lib/std/json/static.zig

```diff
@@ -196,7 +209,7 @@ fn internalParse(
             };
         },
         .Float, .ComptimeFloat => {
-            const token = try source.nextAllocMax(allocator, .alloc_if_needed, options.max_value_len.?);
+            const token = try source.nextAllocMax(allocator, options.allocate, options.max_value_len.?);
```

Comment:
I don't think there is a need to  change the allocation strategy here - the allocation is freed again and is not used by the returned value.

Reply:
Addressed